src/MSBuild/XMake.cs

```diff
@@ -1725,12 +1725,16 @@ internal static void GatherCommandLineSwitches(ArrayList commandLineArgs, Comman
                                 String.Equals(switchName, "maxcpucount", StringComparison.OrdinalIgnoreCase))
                             {
                                 int numberOfCpus = Environment.ProcessorCount;
-                                if (!Environment.Is64BitProcess && NativeMethodsShared.IsWindows) // 32-bit process, 32-bit Windows had a 32-core limit
+#if !RUNTIME_TYPE_NETCORE
```

Comment:
```suggestion
#if NETFRAMEWORK && !MONO
```

I think this is slightly clearer and preserves the intent: do this only on full .NET Framework.